### ISSUE TITLE
Crushed skeletal mesh on import

### ISSUE THREAD

**Issue by CoreyHUN** [poly-hammer/BlenderTools#110]
Hi! 

I ran into a problem with importing skeletal meshes to unreal. Upon import, all the skeletal meshes are crushed/distorted for some reason. 

For representation, I just exported the default UEFN mannequin to blender.

![image](https://github.com/user-attachments/assets/45403927-c087-41b6-ae40-87859bbebc19)

When trying to import it back unreal with **Pipeline/Export/Push** the result is this

![image](https://github.com/user-attachments/assets/c7fd2797-0da6-4bdd-af30-1b80e999718c)

Any idea on this?

I'm using:
Blender 4.3.2 
Unreal 5.5.1
send2ue 2.6.1

**Comment by JoshQuake** [comment]
Hey there. Make sure you apply scale to the skeleton after importing to blender. Also it appears there's an issue with UE5.5's interchange that could also possibly be messing some things up. I hear 5.5.2 should fix the interchange issues.

**Comment by CoreyHUN** [comment]
Hey Josh, yeah I applied the transforms, its gotta be the interchange then, guess I have to wait till .2, thanks!

**Comment by JoshQuake** [comment]
I can also take a quick look to see if the UEFN skele is different from manny. Can you please share the UEFN character? I dont wanna install to get a copy heh.

**Comment by CoreyHUN** [comment]
Hey, no worries, here is the UEFN exported, this is the one I tested with.

https://drive.google.com/file/d/1oiZAtWxsKHGtwtYL0UZ8NOhy0LKPgcHx/view?usp=drive_link

**Comment by JoshQuake** [comment]
thank you!

**Comment by JoshQuake** [comment]
Yup it's all good in 5.3 without doing anything. So this is almost definitely the interchange issue. Should work in .2 🤞

**Comment by CoreyHUN** [comment]
Great! Thank you for checking!

**Comment by liuk997** [comment]
I’ve also encountered the same problem and discovered that in my case it was due to the scene scale.
In Blender, I have the _Unit Scale_ set to 0.01, and when I export with send2ue, the animations work fine, but the skeletal mesh appears collapsed, as described in the post. 
I tried disabling the _Apply Unit_ option and directly setting the _Scale_ to 0.01, but it didn’t work.
However, if I disable the _Apply Unit_ option and set the scale to 0.02, everything works fine (aside from the fact that the scale is, of course, not the desired one). 
0.01 seems to be the threshold where this issue occurs.

I also tried exporting the model directly from Blender in FBX format, setting the scale to 0.01 in Blender's export dialog. In Unreal, the import works with the correct scale and without the collapsed mesh.
I also tried with other meshes, and the problem occurs as well.

I'm using:

Blender 4.4
Unreal 5.3.2
send2ue 2.6.1

**Comment by DanMcLaughlin** [comment]
I found its from applying scale. If you have a mesh parented to a skeleton and Apply Scale to the mesh, or Apply Armature Scale to the rig, Blender will go mess with the other part to maintain the same _relative_ scale - this is what I think is happening at least. What I do is apply the armature scale first, then mesh scale, then double check that both have scale of 1,1,1

**Comment by liuk997** [comment]
I tried several times to apply the scale to both the armature and the mesh, but nothing worked. Both are correctly set to 1 during the export phase. As I mentioned before, if I set the export scale to 0.02 or higher, everything works fine. 
The problem occurs when I use an export scale of 0.01 (only with send2ue export; if I use Blender's classic export to then import the FBX, everything works). 
Could it be caused by numerical precision? Some error propagating during the export that causes a mismatch in the scale of the mesh and the armature? I’m not sure.

**Comment by JoshQuake** [comment]
May I ask why you're changing the export scale at all?

As long as scene scale is correct (metric, 1.0 unit scale) and your objects scales are at 1, then you should get accurate scale in unreal.

**Comment by liuk997** [comment]
I created the mesh and animations using Rigify with a larger scale for the meshes, so when I import them into Unreal, they are too big. I should rescale the animations, but every time I try, something breaks. However, scaling the entire scene during export works, except that the mesh collapses. 
If it helps, I did some testing today and found out that by changing the Forward and Up settings in the export dialog under the Transform section, the issue doesn’t occur (though, of course, the mesh appears rotated in Unreal).

**Comment by jack-yao91** [comment]
Have you tried re-importing the animations from an FBX via the [Import Asset](https://poly-hammer.github.io/BlenderTools/send2ue/extras/pipeline-menu/#import-asset) operator? Its basically an FBX import plus the scale corrections to Skeletons and animations that comes from unreal.

Here is the call it uses to scale the actions
https://github.com/poly-hammer/BlenderTools/blob/fff28ece139ab8e88a56c9b21912c42a0a21eeee/src/addons/send2ue/core/utilities.py#L1572


Could probably fix them in your scene with this; provided the right armature object is selected and the action name is changed.
```python
from send2ue.core.utilities import scale_object_actions
scale_object_actions(
    bpy.context.selected_objects, 
    [bpy.data.actions['MyActionNameHere']], 
   1.0
)
```

Haven't tested that code, but I think that would work to re-scale your actions (maybe play with the last number `1.0`)

**Comment by JoshQuake** [comment]
> I created the mesh and animations using Rigify with a larger scale for the meshes, so when I import them into Unreal, they are too big. I should rescale the animations, but every time I try, something breaks. However, scaling the entire scene during export works, except that the mesh collapses. If it helps, I did some testing today and found out that by changing the Forward and Up settings in the export dialog under the Transform section, the issue doesn’t occur (though, of course, the mesh appears rotated in Unreal).

Something else to consider for ya; Blender's bone system uses different axis from Unreal, which is why the skeleton "fixes" when you change the forward/up axis. UE2Rigify was made to solve this issue to give you a Rigify rig that will bake animations correctly for UE.

It might be worth baking your animations to your skeleton, throw away your manual Rigify rig, and create a new template for your characters with UE2Rigify. You are then able to bake your animations back to the new rig automatically to continue modifying your animations with the new rig.

**Comment by CoreyHUN** [comment]
Great, thank you!

**Comment by JoshQuake** [comment]
To update, 5.5.2 did not fix support for import options. Bug report has been sent.

**Comment by JoshQuake** [comment]
closing out as the flag is good enough for now. When interchange is more fleshed out I will make the changes to properly utilize interchange instead of relying on legacy.

**Comment by jack-yao91** [comment]
If you could post a .blend file so that people can reproduce this, that would be help in getting to the bottom of this
